docs/language/expressions.md

```diff
@@ -81,6 +78,22 @@ For example,
 {d:{e:1}}
 ```
 
+Complex values are recursively traversed to determine if the item is present
+anywhere within them:
+```mdtest-spq
+# spq
+{s:"foo"} in this
+# input
+{s:"foo"}
+{s:"foo",t:"bar"}
+{a:{s:"foo"}}
+[1,{s:"foo"},2]
+# expected output
+{s:"foo"}
+{a:{s:"foo"}}
+[1,{s:"foo"},2]
+```
```

Comment:
I broke out this separate example because the "Search Expressions" doc (that's also touched in this PR) makes a reference to using `in this` with complex values and so I wanted to link to this section of the "Expressions doc", but that's when I noticed we didn't show an example of it here.